crates/proof-of-sql/src/utils/parquet_to_commitment_blob.rs

```diff
@@ -0,0 +1,714 @@
+        .columns()
+        .iter()
+        .map(|c| take(c, &indices, None).unwrap())
+        .collect();
+    RecordBatch::try_new(schema, columns).unwrap()
+}
+
+fn cast_string_array_to_decimal256_array(
+    string_array: &Vec<Option<String>>,
+    precision: u8,
+    scale: i8,
+) -> Decimal256Array {
+    let mut builder =
+        Decimal256Builder::default().with_data_type(DataType::Decimal256(precision, scale));
+
+    string_array.iter().for_each(|value| match value {
+        Some(v) => {
+            let decimal_value = f64::from_str(v).expect("Invalid number");
+            let scaled_value = decimal_value * 10f64.powi(scale as i32);
+            builder.append_value(i256::from_f64(scaled_value).unwrap());
+        }
+        None => builder.append_null(),
+    });
+
+    builder.finish()
+}
+
+fn correct_utf8_fields(
+    record_batch: RecordBatch,
+    big_decimal_columns: Vec<(String, u8, i8)>,
+) -> RecordBatch {
+    let big_decimal_columns_lookup: HashMap<String, (u8, i8)> = big_decimal_columns
+        .into_iter()
+        .map(|(key, precision, scale)| (key, (precision, scale)))
+        .collect();
+    let schema = record_batch.schema();
+
+    // Replace StringArray columns as appropriate
+    let columns: Vec<Arc<dyn Array>> = record_batch
+        .columns()
+        .iter()
+        .zip(schema.fields().iter())
+        .map(|(pointer_column, field)| {
+            let column = pointer_column.clone();
+            let column_name = field.name().to_lowercase();
+            if field.data_type() != &DataType::Utf8 {
+                Arc::new(column)
+            } else {
+                let string_vec: Vec<Option<String>> = column
+                    .as_any()
+                    .downcast_ref::<StringArray>()
+                    .unwrap()
+                    .into_iter()
+                    .map(|s| s.map(|st| st.replace("\0", "")))
+                    .collect();
+                big_decimal_columns_lookup
+                    .get(&column_name)
+                    .map(|(precision, scale)| {
+                        Arc::new(cast_string_array_to_decimal256_array(
+                            &string_vec,
+                            *precision,
+                            *scale,
+                        )) as ArrayRef
+                    })
+                    .unwrap_or(Arc::new(StringArray::from(string_vec)))
+            }
+        })
+        .collect();
+
+    // Replace Utf8 fields with Decimal256 for the big_decimal columns
+    let fields: Vec<Arc<Field>> = schema
+        .fields()
+        .iter()
+        .map(|field| {
+            if field.data_type() == &DataType::Utf8 {
+                big_decimal_columns_lookup
+                    .get(&field.name().to_lowercase())
+                    .map(|(precision, scale)| {
+                        Arc::new(Field::new(
+                            field.name(),
+                            DataType::Decimal256(*precision, *scale),
```

Comment:
Our `precision` for big decimal columns generally exceed Decimal256 max precision which is 76